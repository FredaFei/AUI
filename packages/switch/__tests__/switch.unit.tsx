import * as renderer from 'react-test-renderer'
import React from 'react'
import {shallow, mount} from 'enzyme'

import Switch from '../switch'

describe('测试 Switch 以下行为', () => {
  it('Switch 存在', () => {
    const json = renderer.create(<Switch defaultChecked/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Switch 设置 defaultChecked', () => {
    const wrapper = mount(<Switch defaultChecked/>)
    expect(wrapper.props().defaultChecked).toEqual(true)
    expect(wrapper.state('checked')).toEqual(true)
    expect(wrapper.find('.am-switch').hasClass('am-switch-active')).toEqual(true)
  })
  it('Switch 设置 checked', () => {
    const checked = true
    const wrapper = mount(<Switch checked={checked}/>)
    expect(wrapper.props().checked).toEqual(true)
    expect(wrapper.find('.am-switch').hasClass('am-switch-active')).toEqual(true)
  })
  it('Switch 设置 disabled', () => {
    const fn = jest.fn();
    const wrapper = mount(<Switch disabled defaultChecked onChange={fn}/>)
    expect(wrapper.props().disabled).toEqual(true)
    expect(wrapper.find('.am-switch').hasClass('am-switch-disabled')).toEqual(true)
    wrapper.find('.am-switch').simulate('click')
    expect(fn.mock.calls.length).toEqual(0);
  })
  it('Switch 手动点击，自动触发 state.checked & props.onChange', () => {
    const fn = jest.fn();
    const wrapper = mount(<Switch defaultChecked onChange={fn}/>)
    expect(wrapper.props().defaultChecked).toEqual(true)
    wrapper.find('.am-switch').simulate('click')
    expect(fn.mock.calls.length).toEqual(1);
    expect(wrapper.state('checked')).toEqual(false)
    expect(wrapper.find('.am-switch').hasClass('am-switch-active')).toEqual(false)
  })
  it('Switch 手动触发 onChange', () => {
    const wrapper = mount(<Switch checked={true}/>)
    wrapper.setProps({checked: false});
    expect(wrapper.props().checked).toEqual(false)
    expect(wrapper.find('.am-switch').hasClass('am-switch-active')).toEqual(false)
  })
  it('Switch 设置 自定义类名', () => {
    const wrapper = shallow(<Switch defaultChecked className="test"/>)
    expect(wrapper.find('.am-switch').hasClass('test')).toBe(true)
  })
  it('Switch 设置 自定义样式', () => {
    const wrapper = shallow(<Switch defaultChecked style={{marginRight: '30px'}}/>)
    expect(wrapper.render()).toMatchSnapshot()
  });
})