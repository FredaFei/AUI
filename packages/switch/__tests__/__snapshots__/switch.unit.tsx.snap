// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Switch 以下行为 Switch 存在 1`] = `
<label
  className="am-switch am-switch-active "
  onClick={[Function]}
  style={Object {}}
>
  <span
    className="am-switch-core active"
  >
    <span
      className="am-switch-ripple active"
      style={Object {}}
    />
  </span>
</label>
`;

exports[`测试 Switch 以下行为 Switch 设置 自定义样式 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "am-switch am-switch-active ",
      "style": "margin-right:30px",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "am-switch-core active",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "am-switch-ripple active",
            },
            "children": Array [],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
